### ISSUE TITLE
Can't select items after opening a dialog

### ISSUE THREAD

**Issue by heruan** [vaadin/vaadin-combo-box-flow#161]
After opening a dialog, items are no more selectable.

![freezing-combo-box](https://user-images.githubusercontent.com/851745/49438785-9d058900-f7bf-11e8-8fa2-909f59285d42.gif)

Demo at: [ComboBoxWithDialog.java](https://github.com/heruan/vaadin-spring-demo/blob/072bb0a8118a47fbc8e1c05a21c56b642153af07/src/main/java/to/lova/vaadin/spring/ComboBoxWithDialog.java)

Some details on the specific example:

- the dialog is triggered by a `slot="prefix"` button;
- the dialog is a `ConfirmDialog`.

This is similar to other overlay related issues, but the fixes for them didn't fix this.

I also tried to replicate this with pure web-components, but I wasn't able to (see https://glitch.com/edit/#!/gorgeous-twilight).

**Comment by heruan** [comment]
Any update or workaround about this?

**Comment by moryakovdv** [comment]
Same here.
It seems to me that inside any Layout ComboBox is broken after first selection event. 
Try to use keyboard, not mouse, and the selection event fires(!)  
Also if you put the ComboBox inside the Dialog (or any layout in the Dialog) it works fine.

**Comment by heruan** [comment]
Thank you @moryakovdv for the insight. I hope to hear from the team about this, it's been a while and would be great to have this fixes on V13.

**Comment by moryakovdv** [comment]
@heruan but V13 is far away from now, isn't it?
In complex UIs this problem may became blocking, because the problem concerns the vaadin-overlay, not just Combobox. I suspect  two comboboxes or any dropdown components (such as datepicker) may lead to broken UI.  I hope, the team'll agree to increase the priority.

**Comment by web-padawan** [comment]
We will investigate the root issue based on your feedback in vaadin/vaadin-dialog#123

**Comment by web-padawan** [comment]
Most likely a duplicate of the ticket mentioned above, will fix that one first.

**Comment by moryakovdv** [comment]
By the way, it seems I found a workaround for the Java API. 
The main point is to remove Property('pointer-events') from combobox dropdown.
**Firstly** we should extend the ComboBox to make the `addOpenedChangeListener` method visible:
```java
public class FixedCombobox<T> extends ComboBox<T> {
	private static final long serialVersionUID = 5252096258742402441L;
	public FixedCombobox(String caption) {
		super(caption);
	}
	@Override
	public Registration addOpenedChangeListener(ComponentEventListener<OpenedChangeEvent<ComboBox<T>>> listener) {
		return super.addOpenedChangeListener(listener);
	}
}
```
**Secondly** make the parent component for the combobox implementing the `ComponentEventListener` interface and add it to Combobox listeners, e.g:
```java
public class DocumentsList extends VerticalLayout implements ComponentEventListener<OpenedChangeEvent<ComboBox<DocumentType>>> {
/***/
    protected final FixedCombobox<DocumentType> cbCreateDocument = new FixedCombobox<DocumentType>("Create new");
    public void someInitMethod() {
        cbCreateDocument.addOpenedChangeListener(this);
    }

/**other code*/
}
```
**Thirdly** implement the `onComponentEvent` there:
```java
@Override
public void onComponentEvent(OpenedChangeEvent<ComboBox<DocumentType>> event) { 
    if (event.isOpened())
        getUI().get().getPage().executeJavaScript("Array.from(document.body.children).filter(el => el.shadowRoot!=null && el.id===\"overlay\" ).forEach(o=>(o.shadowRoot.querySelector('[part=\"overlay\"]').style.removeProperty(\'pointer-events\')));");		
}
```
So, when you open combobox all active overlays on the page drops their `pointer-events:none` css property.
I'm not completely sure that this solution has not other issues, but it may help.

**Comment by stanofujdiar** [comment]
@moryakovdv I have same issue is with vaadin-select-flow component but your workaround is not working beacause of little bit different component structure(e.g. there is no overlay id). There are only tagNames like vaadin-select-overlay. But, If I remove pointer-events property manualy (in browser) then select works OK. But I don't know how to write correct selector.

**Comment by moryakovdv** [comment]
I'm not sure it is a right place to discuss it, actually it's an issue tracker, not Stackoverflow :)
But still, try to remove condition ` && el.id===\"overlay\" ` and see for result. In this case any  element with shadowRoot property will be checked for presence of its overlay part and style properties.
If you still have issues try to show me (by email to prevent flood here) your dom-code from firebug and we'll try to see together for workaround.

**Comment by web-padawan** [comment]
> When item in the combobox dropdown list is clicked the dialog window occurs. After the dialog is closed we lost mouse events in dropdown list.

Thanks for the detailed reproduction @moryakovdv 
I have tested it with the `vaadin-overlay` 3.2.6 released yesterday and confirmed that now it works properly, so pointer-events setting and removal timings are correct:
https://glitch.com/edit/#!/fern-bream?path=index.html:11:0

Closing as fixed by vaadin/vaadin-overlay#140

Note: the fix will be shipped in the upcoming V13 beta and the next V12 maintenance release.
